src/virtiofsd.rs

```diff
@@ -0,0 +1,141 @@
+use std::path::Path;
+use std::path::PathBuf;
+use std::process::Child;
+use std::process::Command;
+use std::process::Stdio;
+use std::sync::LazyLock;
+use std::time::Duration;
+use std::time::Instant;
+
+use anyhow::bail;
+use anyhow::Context as _;
+use anyhow::Result;
+use log::error;
+
+use crate::util::gen_sock;
+
+const VIRTIOFSD_NAME: &str = "virtiofsd";
+static VIRTIOFSD_PATH: LazyLock<Option<PathBuf>> = LazyLock::new(find_virtiofsd_exe);
+
+fn find_virtiofsd_exe() -> Option<PathBuf> {
+    // The virtiofsd binary can be found at `/usr/libexec/virtiofsd`
+    // on Debian and Fedora, `/usr/lib/virtiofsd` on Arch.
+    let dirs = ["/usr/libexec/", "/usr/lib/"];
+    for dir in dirs {
+        let path = Path::new(dir).join(VIRTIOFSD_NAME);
+        if path.exists() {
+            return Some(path);
+        }
+    }
+
+    None
```

Comment:
Also check $PATH - this won't work on nixos

Reply:
Those folders are generally not in `PATH`. Is that different on nixos? Where is the daemon located there?

Reply:
Nix does all sorts of special things:

```
dlxu@dlxu-fedora~ $ nix-shell -p virtiofsd

[nix-shell:~]$ which virtiofsd
/nix/store/mm5dj9nm4bm0d238mln61q0vbkasjh9c-virtiofsd-1.11.1/bin/virtiofsd
```

basically everything is content addressed. and for discovery/consumption everything executable is on $PATH

Reply:
Okay, thanks, easy to add given how it's currently written.